File: IT/1/Variabili, operatori, strutture di controllo/C++/testRoundingError.cpp
/* file testRoundingError.cpp */

#include <iostream>
#include <iomanip>

using std::cout;
using std::endl;
using std::setprecision;

int main() {
	double x = 0.1;
	double y = 1.0;

	for (int i = 1; i < 10; i++) {
		cout << setprecision(16) << x << endl;
		x += 0.1;
	}

	if (x == y) {
		cout << x << " e " << y << " sono uguali!" << endl;
	}
	else {
		cout << x << " e " << y << " sono diversi!" << endl;
	}
}

File: IT/1/Variabili, operatori, strutture di controllo/Java/TestRoundingError.java
/* file TestRoundingError.java */

public class TestRoundingError {
	public static void main(String [] args) {
		double x = 0.1;
		double y = 1.0;

		for (int i = 1; i < 10; i++) {
			System.out.println(x);
			x += 0.1;
		}

		if (x == y) {
			System.out.println(x + " e " + y + " sono uguali!");
		}
		else {
			System.out.println(x + " e " + y + " sono diversi!");
		}
	}
}

